src/extensions/lens-main-extension.ts

```diff
@@ -33,6 +34,19 @@ export class LensMainExtension extends LensExtension {
     return WindowManager.getInstance().navigateExtension(this.id, pageId, params, frameId);
   }
 
+  /**
+   * Add a filtering function for the catalog. This will be removed if the extension is disabled.
+   * @param fn The function which should return a truthy value for those entities which should be kepted
+   * @returns A function to clean up the filter
+   */
+  addCatalogFilter(fn: EntityFilter): Disposer {
```

Comment:
I'm not sure if this is the correct place for filters. I would thought that filter only applies to visible items, not to whole catalog data. Doing filtering at this level might cause unintended side-effects to other extensions.

Reply:
My suggestion: let's do filtering on renderer side, maybe only for Catalog UI.

Reply:
Sure I guess, I would have figured that this sort of filtering was supposed to be pervasive. That way it actually feels like switching.